pkg/autoscaler/autoscaler.go

```diff
@@ -291,6 +294,106 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+			}
+
+			By("Creating a new MachineSet with 0 replicas for each machineset of a different architecture found in the cluster")
+			expectedReplicas := int32(1)
+			machineSetParamsList := framework.BuildPerArchMachineSetParamsList(ctx, client, 0)
+			machineSets := make([]*machinev1.MachineSet, 0)
+			targetedNodeLabel := fmt.Sprintf("%v-scale-from-zero", autoscalerWorkerNodeRoleLabel)
+			var expectedScaledMachineSet *machinev1.MachineSet
+			var workloadArch string
+
+			for _, machineSetParams := range machineSetParamsList {
+				machineSetParams.Labels[targetedNodeLabel] = ""
+				By(fmt.Sprintf("Deploying the MachineSet with 0 replicas (architecture: %s)",
+					machineSetParams.Labels[framework.ArchLabel]))
+				machineSet, err := framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred(), "Failed to create MachineSet with 0 replicas")
+				machineSets = append(machineSets, machineSet)
+				cleanupObjects[machineSet.GetName()] = machineSet
+
+				framework.WaitForMachineSet(ctx, client, machineSet.GetName())
+
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s/%s - min:%v, max:%v",
+					machineSet.GetNamespace(), machineSet.GetName(), 0, expectedReplicas))
+				asr := machineAutoscalerResource(machineSet, 0, expectedReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), fmt.Sprintf("Failed to create MachineAutoscaler with min 0/max %v replicas", expectedReplicas))
+				cleanupObjects[asr.GetName()] = asr
+				arch := machineSetParams.Labels[framework.ArchLabel]
+				if arch != framework.Amd64 || workloadArch == "" {
+					// the workloadArch == "" condition ensures that workloadArch and expectedScaledMachineSet are set
+					// at least once. This is needed to ensure that the test does not fail when there is only one amd64 MachineSet
+					workloadArch = arch
+					expectedScaledMachineSet = machineSet
+				}
+			}
+			uniqueJobName := fmt.Sprintf("%s-scale-from-zero", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s", uniqueJobName,
+				expectedReplicas+1, workloadMemRequest.String()))
+			// Executing one additional replica to ensure that the workload does not triggers other scale-out events with
+			// other MachineSets not having the requested architecture.
+			workload := framework.NewWorkLoad(expectedReplicas+1, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "",
+				corev1.NodeSelectorRequirement{
+					Key:      targetedNodeLabel,
+					Operator: corev1.NodeSelectorOpExists,
+				},
+				corev1.NodeSelectorRequirement{
+					Key:      "kubernetes.io/arch",
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{workloadArch},
+				})
+
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", workloadJobName)
+			passed := framework.RunCheckUntil(ctx,
```

Comment:
The context passed in here should be a context with timeout to give a time for the check to finish at, so use `runCtx, cancel := context.WithTimeout(...)` to give the run a time limit